File: back/src/main/java/com/julien/plop/player/domain/model/Player.java
package com.julien.plop.player.domain.model;

import com.julien.plop.i18n.domain.Language;
import com.julien.plop.tools.StringTools;

import java.util.Objects;

public record Player(Id id, String name, Language language) {

    public record Id(String value) {
        public Id() {
            this(StringTools.generate());
        }
    }

    public Player(String name) {
        this(new Id(), name, Language.first());
    }

    @Override
    public boolean equals(Object o) {
        if (!(o instanceof Player player)) return false;

        return Objects.equals(id, player.id);
    }

    @Override
    public int hashCode() {
        return Objects.hashCode(id);
    }
}

File: mobile/lib/contexts/player/player.dart

import 'package:plop/generic/config/language.dart';

class Player {

  final String _id;
  final String _name;
  final Language _language;

  Player({
    required String id,
    required String name,
    required Language language
  }) : _id = id, _name = name, _language = language;

  String get id => _id;

  String get name => _name;

  Language get language => _language;

}